src/components/ui/optionsPopover/OptionsPopover.tsx

```diff
@@ -47,23 +31,11 @@ export const OptionDialogItem = ({
   ButtonElement: ReactNode;
 }) => {
   const { handleClose: closePopover } = useContext(PopoverContext);
-  const [open, setOpen] = useState(false);
-
-  const handleClose = () => setOpen(false);
-
   return (
-    <OptionsDialogContext.Provider value={{ handleClose }}>
-      <Dialog
-        onOpenChange={() => {
-          closePopover();
-          setOpen(!open);
-        }}
-        open={open}
-      >
-        <DialogTrigger asChild>{ButtonElement}</DialogTrigger>
-        {DialogElement}
-      </Dialog>
-    </OptionsDialogContext.Provider>
+    <DialogBase onOpenChange={() => closePopover()}>
+      <DialogTrigger asChild>{ButtonElement}</DialogTrigger>
+      {DialogElement}
```

Comment:
actually -- can we make this say ButtonComponent and DialogComponent
think its more clear that way